DEHPEcosimPro/ViewModel/Rows/ParameterDifferenceViewModel.cs

```
@@ -0,0 +1,324 @@
+            }
+            else
+            {
+                NewValue = setOfNewValues.FirstOrDefault();
+            }
+
+            if (setOfOldValues.Count > 1)
+            {
+                OldValue = setOfOldValues.ToString();
+            }
+            else
+            {
+                OldValue = setOfOldValues.FirstOrDefault();
+            }
+
+
+            if (isOptions && isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}\\{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (isOptions && !isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}";
+            }
+            else if (!isOptions && isState)
+            {
+                Name = Name + $"{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (!isOptions && !isState)
+            {
+                Name = this.ModelCode();
+            }
+
+            this.CalculateDiff(OldValue, NewValue, out string Difference, out string PercentDiff);
+
+            return new ParameterDifferenceRowViewModel(this.OldThing, this.NewThing, Name, OldValue, NewValue, Difference, PercentDiff);
+
+        }
+
+        /// <summary>
+        /// Add The valueSet to list
+        /// </summary>
+        /// <param name="option"><see cref="Option"/></param>
+        /// <param name="state"><see cref="ActualFiniteState"/></param>
+        private void PopulateListOfSets(Option option, ActualFiniteState state)
+        {
+            try
+            {
+                this.listofsetOfOldValues.Add(this.OldThing.QueryParameterBaseValueSet(option, state));
+                this.listofsetOfNewValues.Add(this.NewThing.QueryParameterBaseValueSet(option, state));
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
```

Does it need to throw an exception here? If so, please add a log entry before re-thowing, and throw a more specific exception e.g. ArgumentNullException with a message